pkg/apiserver/apiserver.go

```diff
@@ -370,7 +371,24 @@ func write(statusCode int, apiVersion string, codec runtime.Codec, object runtim
 		io.Copy(writer, out)
 		return
 	}
-	writeJSON(statusCode, codec, object, w, isPrettyPrint(req))
+	exporter, exact := getExporter(req)
+	exportJSON(statusCode, codec, object, w, isPrettyPrint(req), exporter, exact)
+}
+
+func getExporter(req *http.Request) (export.Exporter, bool) {
+	exp := req.URL.Query().Get("export")
```

Comment:
Swagger supports specifying query parameters, though we don't use it consistently.

One could view this as a specialized form of field selection.

Probably every resource that supports GET should support export, even if it's a no-op.